src/store/modules/ModDownloadModule.ts

```diff
@@ -0,0 +1,97 @@
+import { ActionTree, GetterTree } from 'vuex';
+
+import StatusEnum from '../../model/enums/StatusEnum';
+import { ImmutableProfile } from '../../model/Profile';
+import ThunderstoreCombo from '../../model/ThunderstoreCombo';
+import ThunderstoreMod from '../../model/ThunderstoreMod';
+import ThunderstoreVersion from '../../model/ThunderstoreVersion';
+import ThunderstoreDownloaderProvider from '../../providers/ror2/downloading/ThunderstoreDownloaderProvider';
+
+import { State as RootState } from '../index';
+
+interface Progress {
+    progress: number;
+    status: number; // StatusEnum
+}
+
+interface ModProgressItem {
+    mod: ThunderstoreMod;
+    version: ThunderstoreVersion;
+    installProgress: Progress;
+    downloadProgress: Progress;
+}
+
+interface State {
+    allDownloads: ModProgressItem[];
+}
+
+/**
+ * State for Mod Downloadning/Installation.
+ */
+export default {
+    namespaced: true,
+
+    state: (): State => ({
+        allDownloads: [],
+    }),
+
+    getters: <GetterTree<State, RootState>>{
+        allDownloads(state): ModProgressItem[] {
+            return state.allDownloads;
+        },
+        activeDownload(state): ModProgressItem {
+            return state.allDownloads.slice(-1)[0]; // Last element of the array
+        },
+        activeDownloadProgress(state): number | undefined {
+            if(state.allDownloads.length > 0) {
+                return state.allDownloads.slice(-1)[0].downloadProgress.progress; // Last element of the array
+            }
+        },
+    },
+    mutations: {},
+    actions: <ActionTree<State, RootState>>{
+
+        //TODO: Do the installation
+        async downloadAndInstallMod(
+            {dispatch, getters, state},
+            params: {
+                profile: ImmutableProfile,
+                mod: ThunderstoreMod,
+                version: ThunderstoreVersion,
+            }
+        ): Promise<boolean> {
+            state.allDownloads.push({
+                mod: params.mod,
+                version: params.version,
+                installProgress: { progress: 0, status: StatusEnum.PENDING },
+                downloadProgress: { progress: 0, status: StatusEnum.PENDING }
+            });
+
+            return new Promise((resolve, reject) => {
+                ThunderstoreDownloaderProvider.instance.download(
+                    params.profile,
+                    params.mod,
+                    params.version,
+                    true,
+                    (progress, modName, status, err) => {
+                        const index = state.allDownloads.findIndex((modProgressItem) => modProgressItem.mod.getName() === modName);
```

Comment:
`findIndex` will return the first matching element, but active download is assumed to be the last item in the array. So if same mod is downloaded twice (e.g. download failure or reinstall after uninstallation), this will update wrong ProgressItem. Either reverse the list before searching, or change the approach so `allDownloads` is "newest first" (which might come handy in DownloadMonitor component).